api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -486,3 +505,108 @@ model HcpOpenShiftClusterAdminCredential {
  *   End HCP cluster credentials
  * =======================================
  */
+
+/*
+ * =======================================
+ *  ExternalAuth resources
+ * =======================================
+ */
+
+/** External Auth profile */
+model ExternalAuthProperties {
+  /** Provisioning state */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ExternalAuthProvisioningState;
+
+  /** Token Issuer profile */
+  issuer: TokenIssuerProfile;
+
+  /** External Auth clients */
+  @maxItems(20)
+  clients?: ExternalAuthClientProfile[];
+
+  /** External Auth claim */
+  claim: ExternalAuthClaimProfile;
+}
+
+/** Token issuer profile */
+model TokenIssuerProfile {
+  /** The URL of the token issuer */
+  url: url;
+
+  /** The audience of the token issuer */
+  @minItems(1)
+  audiences: string[];
+
+  /** The issuer of the token */
+  ca?: string;
+}
+
+/** External Auth client profile */
+model ExternalAuthClientProfile {
+  /** External Auth client component */
+  component?: ExternalAuthClientComponentProfile;
+
+  /** external Auth client id */
+  id: string;
+
+  /** external auth client scopes */
+  extraScopes?: string[];
+}
+
+/** External Auth component profile */
+model ExternalAuthClientComponentProfile {
+  /** The name of the external Auth client */
+  name?: string;
+
+  /** The namespace of the external Auth client */
+  authClientNamespace?: string;
```

Comment:
is there a reason why the prefix of `authClient` was added for namespace while `name` doesn't have any?